src/components/cleaner.ts

```diff
@@ -38,6 +62,48 @@ export class Cleaner {
         }
     }
 
+    /**
+     * Splits the given glob patterns into three distinct groups (duplicates will be ignored)
+     *   1. file globs (not end with tailing slashes)
+     *   2. globs explicitly for folders
+     *   3. folder globs with globstar (`**`)
+     *
+     * We will remove the <1.> type paths if they are files, remove the <2.> type
+     * paths if they are empty folders, and ignore the <3.> type paths.
+     *
+     * @param globs a list of glob patterns
+     * @returns three distinct groups of glob patterns
+     */
+    private static splitGlobs(globs: string[]): { fileGlobs: string[], folderGlobsExplicit: string[], folderGlobsWithGlobstar: string[] } {
+        const fileGlobs: string[] = []
+        const folderGlobsWithGlobstar: string[] = []
+        const folderGlobsExplicit: string[] = []
+
+        for (const pattern of unique(globs)) {
+            if (pattern.endsWith(path.sep)) {
+                // The pattern ends with a slash only matches folders
+                if (path.basename(pattern).includes('**')) {
+                    // Contains globstar `**` in the last component
+                    folderGlobsWithGlobstar.push(pattern)
+                } else {
+                    // All glob patterns EXPLICITLY given for folders, the glob should be end with a slash (`path.sep`)
+                    // and the last component should not contain globstar `**`
+                    //
+                    // Positive examples: ['abc/', 'abc*/', '**/abc*/', 'abc/**/def/', 'abc/**/def*/']
+                    // Negative examples:
+                    //   - not end with a slash: ['abc', 'abc*', 'abc/**/def*']
+                    //   - contain globstar `**` in the last component: ['**', '**/', 'abc/**', 'abc/**/', 'abc/def**/', 'abc/d**ef/']
```

Comment:
Move these explanations in the documentation of the function. Long comments tend make the code hard to read and become counterproductive.

Reply:
Glob pattern `*/` cannot feed into multi-line comment `/** ... */`.